lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -1097,35 +1096,19 @@ public function getSchemaSearchPaths()
     }
 
     /**
-     * Given a table comment this method tries to extract a typehint for Doctrine Type, or returns
-     * the type given as default.
+     * Given a table comment this method tries to extract a type hint for Doctrine Type. If the type hint is found,
+     * it's removed from the comment.
      *
-     * @param string|null $comment
-     * @param string      $currentType
-     *
-     * @return string
-     */
-    public function extractDoctrineTypeFromComment($comment, $currentType)
-    {
-        if ($comment !== null && preg_match('(\(DC2Type:(((?!\)).)+)\))', $comment, $match)) {
-            return $match[1];
-        }
-
-        return $currentType;
-    }
-
-    /**
-     * @param string|null $comment
-     * @param string|null $type
-     *
-     * @return string|null
+     * @return string|null The extracted Doctrine type or NULL of the type hint was not found.
      */
-    public function removeDoctrineTypeFromComment($comment, $type)
+    protected function extractDoctrineTypeFromComment(?string &$comment) : ?string
```

Comment:
For those who don't like arguments passed by reference, in this case, it's not much different than `array_shift()`.

Reply:
I'd still rather not have them. What's the alternative that warrants such an API?

Reply:
I don't understand the question. Could you rephrase?

Reply:
The need for the API change was discovered during the introduction of type hints. An alternative might be to return a tuple of `($comment, $type)` which is bad from the static analysis standpoint or to have two methods which is weird because you'd do one thing in two steps.

There's nothing wrong about the comment being passed by reference and mutated. That's why I made it protected. A caller doesn't need to keep the unmodified version of the comment for future use since the Doctrine type hint contained in the original value is internal to DBAL and just should be removed from it.

I do understand the implications of passing arguments by reference but IMO it's justified in this case.

Reply:
I don't like the side-effect of this method - the name is "extract" but it also removes the type part from the comment. This may or may not be expected by the caller.
Maybe at least using separate parameter, like `&$filteredComment` would be a good idea?

Reply:
> I don't like the side-effect of this method - the name is "extract" but it also removes the type part from the comment. This may or may not be expected by the caller.

In my understanding, "extract" means exactly what the method does (think of a tooth or an intelligence asset extraction). You have the object of extraction as the result, and it's no longer where it was.

> Maybe at least using separate _out_ parameter, like `&$filteredComment = null` would be a good idea?

There's no need for that. See above:

> A caller doesn't need to keep the unmodified version of the comment for future use since the Doctrine type hint contained in the original value is internal to DBAL and just should be removed from it.

Reply:
> You have the object of extraction as the result, and it's no longer where it was.

Hmm, it makes sense. Maybe it's just me but when a method is called _extract_, I'd expect the portion being returned and the input untouched.

Reply:
> I'd expect the portion being returned and the input untouched.

I'd expect exactly the opposite:

![image](https://user-images.githubusercontent.com/59683/55520601-ee206f00-5631-11e9-8f95-a366c2346840.png)

It literally means "remove".

Reply:
> Maybe at least using separate _out_ parameter, like `&$filteredComment = null` would be a good idea?

It makes the API just harder to use since the caller will eventually have to replace their original `$comment` with `$filteredComment` which the proposed API already does for them. Look at the calling code.